### PR TITLE
Add "Best Meeting Point in Manhattan / Grid-Like City" Question.

### PR DESCRIPTION
**Pull request by cwcai633**
Added Q5 "find best meeting point in grid-like city".

First, the candidate is asked to give any one answer from all the correct answers.

Second, the candidate is asked to give all possible answers.

### CODE REVIEW

File: assets/question_sets/strings.js [hunk 1]
```diff
@@ -26,5 +26,5 @@ globalData.questionSets['strings'] = {  // eslint-disable-line dot-notation
       "I will provide you with feedback."
     ].join('')
   ],
-  questionIds: ['reverseWords', 'parens', 'i18n', 'rle']
```
Comment (eyurko) [inline assets/question_sets/strings.js]: Please see the email from this morning -- no need to add these questions to strings.js for the final commit.

File: assets/questions/manhattan.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Question data for Best Meeting location for N People in
+ * a Manhattan-like City.
+ */
+
+globalData.questions['manhattan'] = {  // eslint-disable-line dot-notation
+  title: 'Best Meeting location for N People in a Manhattan-like City',
```
Comment (eyurko) [inline assets/questions/manhattan.js]: Location

File: assets/questions/manhattan.js [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+        'N people live in a city with streets that form a regular 2D grid. ',
+        'They start out at different intersections and want to meet up. ',
+        'They are only able to move along the roads (no crossing blocks diagonally etc.). ',
+        'Find the meeting place that minimizes the total travel distance.'
+      ].join(''),
+      [
+        'The input is a str list indicating the coordinates of the people in a 2D grid. ',
+        'Your function should return a str indicating the best meeting point for N people. ',
+        'For instance, the best meeting point for 3 people at ',
+        '"0, 0", "4, 2", "2, 4" is "2, 2".'
```
Comment (eyurko) [inline assets/questions/manhattan.js]: and "2, 4"

File: assets/questions/manhattan.js [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+      ].join('')
+    ],
+    prerequisiteSkills: ['Math', 'Sort'],
```
Comment (eyurko) [inline assets/questions/manhattan.js]: "Sorting"

ditto below

File: assets/questions/manhattan.js [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+        return None
+`
+  },
+  tasks: [{
+    instructions: [
+      [
+        'N people live in a city with streets that form a regular 2D grid. ',
+        'They start out at different intersections and want to meet up. ',
+        'They are only able to move along the roads (no crossing blocks diagonally etc.). ',
+        'Find the meeting place that minimizes the total travel distance.'
```
Comment (eyurko) [inline assets/questions/manhattan.js]: Write a function manhattan(people) that returns a meeting place that minimizes the total travel distance.

File: assets/questions/manhattan.js [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+    }, {
+      input: [],
+      allowedOutputs: ['']
+    }],
+    buggyOutputTests: [],
+    performanceTests: []
+  }, {
+    instructions: [
+      [
+        'Can you find all possible best meeing points? ',
```
Comment (eyurko) [inline assets/questions/manhattan.js]: meeting

File: assets/questions/manhattan.js [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+        '(Return List[str] with function manhattan)'
```
Comment (eyurko) [inline assets/questions/manhattan.js]: This seems incomplete. May I suggest:

"You might notice that some inputs have more than one possible "best" meeting point. Can you modify your manhattan function to return a list of all of the best meeting points?

File: assets/question_sets/strings.js (hunk 1 above)
Comment (cwcai633) [inline assets/question_sets/strings.js]: Done.

File: assets/questions/manhattan.js (hunk 2 above)
Comment (cwcai633) [inline assets/questions/manhattan.js]: Done.

File: assets/questions/manhattan.js (hunk 5 above)
Comment (cwcai633) [inline assets/questions/manhattan.js]: Done.

File: assets/questions/manhattan.js (hunk 3 above)
Comment (cwcai633) [inline assets/questions/manhattan.js]: Done.

File: assets/questions/manhattan.js (hunk 4 above)
Comment (cwcai633) [inline assets/questions/manhattan.js]: Done.

File: assets/questions/manhattan.js (hunk 6 above)
Comment (cwcai633) [inline assets/questions/manhattan.js]: Done.

File: assets/questions/manhattan.js (hunk 7 above)
Comment (cwcai633) [inline assets/questions/manhattan.js]: Done.

File: client/domain/CorrectnessTestObjectFactory.js [hunk 8]
```diff
@@ -31,7 +31,17 @@ tie.factory('CorrectnessTestObjectFactory', [
     };
 
     CorrectnessTest.prototype.matchesOutput = function(output) {
-      return this._allowedOutputs.indexOf(output) !== -1;
+      if (typeof this._allowedOutputs === 'object') {
```
Comment (eyurko) [inline client/domain/CorrectnessTestObjectFactory.js]: Are you checking if this is an array? If so, use angular.isArray.

**Comment by seanlip** [comment]
(Please resolve merge conflicts.)

File: assets/questions/manhattan.js [hunk 9]
```diff
@@ -0,0 +1,119 @@
...
+/**
+ * @fileoverview Question data for Best Meeting Location for N People in
+ * a Manhattan-like City.
+ */
+
+globalData.questions['manhattan'] = {  // eslint-disable-line dot-notation
+  title: 'Best Meeting Location for N People in a Manhattan-like City',
+  starterCode: {
+    python:
+`def manhattan(people):
```
Comment (cwcai633) [inline assets/questions/manhattan.js]: Done.

File: assets/questions/manhattan.js [hunk 10]
```diff
@@ -0,0 +1,119 @@
...
+    return ""
+`
+  },
+  auxiliaryCode: {
+    python:
+`class AuxiliaryCode(object):
+    @classmethod
+    def takeTheAverage(cls, people):
```
Comment (cwcai633) [inline assets/questions/manhattan.js]: Done.

File: assets/questions/manhattan.js [hunk 11]
```diff
@@ -0,0 +1,119 @@
...
+        num = len(people)
```
Comment (cwcai633) [inline assets/questions/manhattan.js]: Done.

File: assets/questions/manhattan.js [hunk 12]
```diff
@@ -0,0 +1,119 @@
...
+`
+  },
+  tasks: [{
+    instructions: [
+      [
+        'N people live in a city with streets that form a regular 2D grid. ',
+        'They start out at different intersections and want to meet up. ',
+        'They are only able to move along the roads (no crossing blocks diagonally etc.). ',
+        'Find the meeting place that minimizes the total travel distance. ',
+        'Write a function manhattan(people) that returns a meeting place that minimizes the total travel distance.'
```
Comment (cwcai633) [inline assets/questions/manhattan.js]: Done.

File: assets/questions/manhattan.js [hunk 13]
```diff
@@ -0,0 +1,119 @@
...
+      ].join(''),
+      [
+        'The input is a str list indicating the coordinates of the people in a 2D grid. ',
```
Comment (cwcai633) [inline assets/questions/manhattan.js]: Done.

File: assets/questions/manhattan.js [hunk 14]
```diff
@@ -0,0 +1,119 @@
...
+        'Your function should return a str indicating the best meeting point for N people. ',
```
Comment (cwcai633) [inline assets/questions/manhattan.js]: Done.

File: assets/questions/manhattan.js [hunk 15]
```diff
@@ -0,0 +1,119 @@
...
+      input: ['0, 0', '5, 6', '4, 5', '8, 9'],
+      allowedOutputs: ['4, 5', '4, 6', '5, 5', '5, 6']
+    }, {
+      input: [],
+      allowedOutputs: ['']
+    }],
+    buggyOutputTests: [{
+      buggyFunctionName: 'AuxiliaryCode.takeTheAverage',
+      messages: [
+        "The best meeting point is not necessarily the centroid of all the people. ",
```
Comment (cwcai633) [inline assets/questions/manhattan.js]: Done.

Comment (cwcai633) [inline assets/questions/manhattan.js]: Done.

File: assets/questions/manhattan.js [hunk 16]
```diff
@@ -0,0 +1,119 @@
...
+        "Could that be somewhere else?"
+      ]
+    }],
+    performanceTests: []
+  }, {
+    instructions: [
+      [
+        'You might notice that some inputs have more than one possible "best" meeting point. ',
```
Comment (cwcai633) [inline assets/questions/manhattan.js]: Done.

File: assets/questions/manhattan.js [hunk 17]
```diff
@@ -0,0 +1,119 @@
...
+      allowedOutputs: [['2, 2']]
+    }, {
+      input: ['0, 0', '5, 6', '4, 5', '8, 9'],
+      allowedOutputs: [['4, 5', '4, 6', '5, 5', '5, 6']]
+    }, {
+      input: [],
+      allowedOutputs: ['']
+    }],
+    buggyOutputTests: [],
+    performanceTests: []
```
Comment (cwcai633) [inline assets/questions/manhattan.js]: Done.

File: client/domain/CorrectnessTestObjectFactory.js (hunk 8 above)
Comment (cwcai633) [inline client/domain/CorrectnessTestObjectFactory.js]: Done.

File: assets/question_sets/strings.js (hunk 1 above)
Comment (cwcai633) [inline assets/question_sets/strings.js]: Done again.

File: assets/questions/bestMeetupLocation.js [hunk 18]
```diff
@@ -0,0 +1,135 @@
...
+        return str(x) + ", " + str(y)
+`
+  },
+  tasks: [{
+    instructions: [
+      [
+        'N people live in a city with streets that form a regular 2D grid. ',
+        'They start out at different intersections and want to meet up. ',
+        'They are only able to move along the roads (no crossing blocks diagonally etc.). ',
+        'Find the meeting place that minimizes the total travel distance. ',
```
Comment (eyurko) [inline assets/questions/bestMeetupLocation.js]: This sentence isn't necessary -- it's essentially restated by the next sentence.

File: assets/questions/bestMeetupLocation.js [hunk 19]
```diff
@@ -0,0 +1,135 @@
...
+      allowedOutputs: ['4, 5', '4, 6', '5, 5', '5, 6']
+    }, {
+      input: [],
+      allowedOutputs: ['']
+    }],
+    buggyOutputTests: [{
+      buggyFunctionName: 'AuxiliaryCode.findAverageLocation',
+      messages: [
+        [
+          "Try to run on test case ['0, 0', '4, 2', '10, 10']. ",
```
Comment (eyurko) [inline assets/questions/bestMeetupLocation.js]: D.1

File: assets/questions/bestMeetupLocation.js [hunk 20]
```diff
@@ -0,0 +1,135 @@
...
+          "Is the result what you expected?"
+        ].join(''),
+        [
+          "The best meeting point is not necessary to be ",
```
Comment (eyurko) [inline assets/questions/bestMeetupLocation.js]: D.1

File: assets/questions/bestMeetupLocation.js (hunk 18 above)
Comment (cwcai633) [inline assets/questions/bestMeetupLocation.js]: Done.

File: assets/questions/bestMeetupLocation.js (hunk 19 above)
Comment (cwcai633) [inline assets/questions/bestMeetupLocation.js]: Done.

File: assets/questions/bestMeetupLocation.js (hunk 20 above)
Comment (cwcai633) [inline assets/questions/bestMeetupLocation.js]: Done.

**Pull request merged**